pkg/controller/metrics/metrics.go

```diff
@@ -225,6 +235,25 @@ func (mc *Calculator) Start(ctx context.Context) error {
 						mcLog.Infof("cleared metric: %v", metricClusterDeploymentSyncsetPaused)
 					}
 				}
+
+				// Check if cluster is transitioning
+				hibernatingCond := controllerutils.FindClusterDeploymentCondition(cd.Status.Conditions, hivev1.ClusterHibernatingCondition)
+				readyCond := controllerutils.FindClusterDeploymentCondition(cd.Status.Conditions, hivev1.ClusterReadyCondition)
+				poolRef := ""
+				if cd.Spec.ClusterPoolRef != nil {
+					poolRef = cd.Spec.ClusterPoolRef.PoolName
+				}
+				if hibernatingCond.Reason == hivev1.ResumingOrRunningHibernationReason &&
+					readyCond.Reason != hivev1.RunningReadyReason {
+					metricClusterPowerStateTransitionSeconds.WithLabelValues(cd.Labels[constants.VersionMajorMinorPatchLabel],
+						getCloudPlatform(cd.Spec.Platform),
+						poolRef).Observe(time.Since(hibernatingCond.LastTransitionTime.Time).Seconds())
```

Comment:
I have worries regarding the clearing of metric - while it is possible to find and clear it for corresponding clusterpool, what about the other cds?

Reply:
We're still [discussing](https://coreos.slack.com/archives/CE3ETN3J8/p1641851588051600) this.

My current thinking is: don't clear anything, because it would be really hard to figure out the set of labels you would have to specify.

This would mean that you could pollute new pool "foo" with metrics from old, deleted pool "foo". IMO that's the lesser evil. Document it in the help text and call it good.